src/main/java/com/epam/controller/ToursController.java

```diff
@@ -66,15 +69,24 @@ public ModelAndView searchTours(@RequestParam String country, @RequestParam Stri
                                     @RequestParam String endDate, @RequestParam String numberOfPeople) {
         ModelAndView toursModel = new ModelAndView();
         toursModel.setViewName("homepage");
+        LocalDate addStartDate = null;
+        LocalDate addEndDate = null;
         try {
-            LocalDate addStartDate = Validator.getDate(startDate, true);
-            LocalDate addEndDate = Validator.getDate(endDate, true);
-            toursModel.addObject("hotels", hotelService.getMapOfHotels());
-            toursModel.addObject("list", toursOfferService.searchTours(country, addStartDate, addEndDate));
-            return toursModel;
-        } catch (Exception e) {
-            toursModel.addObject("error", e.getMessage());
-            return toursModel;
+            if(!startDate.isEmpty()) {
```

Comment:
I think u can collapse it. For example, don't allocate new vars. Inline in method's parameters `Date.valueOf() `. If statements here unnecessary, 'cause u have `try-catch`. Just put your code after `catch `inside that block and add `IllegalArgsExeption`.

Reply:
Done